Lib/test/test_sqlite3/test_dbapi.py

```diff
@@ -650,6 +665,16 @@ def __getitem__(slf, x):
         with self.assertRaises(ZeroDivisionError):
             self.cu.execute("select name from test where name=?", L())
 
+    def test_execute_too_many_params(self):
+        category = sqlite.SQLITE_LIMIT_VARIABLE_NUMBER
+        msg = "too many SQL variables"
+        with cx_limit(self.cx, category=category, limit=2):
+            self.cu.execute("select * from test where name=? and income=?",
+                            ("a", 1))
+            with self.assertRaisesRegex(sqlite.OperationalError, msg):
+                self.cu.execute("insert into test values(?, ?, ?)",
```

Comment:
It raises OperationalError even without the limit set.

Reply:
Yes, but an OperationalError with a different error message (it's `assertRaisesRegex`, not `assertRaises`). However, I agree that we should try to raise as accurate as possible. Thanks for your suggested change!